src/crimson/osd/pg_backend.h

```diff
@@ -137,11 +137,18 @@ class PGBackend
 
   virtual void got_rep_op_reply(const MOSDRepOpReply&) {}
   virtual seastar::future<> stop() = 0;
+  struct pg_change_info {
+    bool is_primary;
+  };
+  virtual void on_actingset_changed(pg_change_info pi) = 0;
+  virtual void on_activate_complete();
 protected:
   const shard_id_t shard;
   CollectionRef coll;
   crimson::os::FuturizedStore* store;
   bool stopping = false;
+  bool peering = false;
```

Comment:
how about:
```c++
struct peering_info_t {
  bool is_primary;
};
std::optional<peering_info_t> peering;
```
?

the reason why i don't like `pg_change_info pci` in this change is

1. `pci` only makes sense when `peering` is true
1. the variable name of `pci` is difficult to understand at the first glance.